torchdrivesim/traffic_controls.py

```diff
@@ -191,4 +286,23 @@ class StopSignControl(BaseTrafficControl):
     Stop sign, indicating the vehicle should stop and yield to cross traffic.
     Violations are not computed.
     """
-    pass
+    def compute_violation(self, agent_state: Tensor, time: int) -> Tensor:
+        batch_size, num_agents = agent_state.shape[0], agent_state.shape[1]
+        num_lights = self.corners.shape[1]
+        if batch_size > 0 and num_agents > 0 and num_lights > 0:
+            agent_corners = box2corners_with_rear_factor(agent_state)
+            agent_corners = agent_corners.reshape(-1, 1, 4, 2).expand(-1, num_lights, -1, -1).cuda()
```

Comment:
```suggestion
            agent_corners = agent_corners.reshape(-1, 1, 4, 2).expand(-1, num_lights, -1, -1).to(agent_state)
```
and similarly for others